### ISSUE TITLE
Accessibility: IPython - Microsoft Sponsor page

### ISSUE THREAD

**Issue by bumblebib** [jupyterhub/outreachy#112]
### What page is this for?

https://ipython.org/microsoft-donation-2013.html

### WAVE accessibility report

# Wave Accessibility report: 
![image](https://user-images.githubusercontent.com/90845861/197290024-0fdf3a87-52a9-4003-9192-6ca8988bff7d.png)

The overall issue with the page is that it doesn't fit with the present HTML5 semantics and structure. Among some issues found are: 
- The lang tag not being declared;
- Meta viewport tag missing for a responsive layout; 
- Contrast issues between the link colors and the white background;
- Justified Text that can cause awkward word spacing and word breaking; 
- Images with no alt text, or not descriptive enough; 
- Aria labeled being used for the side menu;
- Side content being declared before the main content;

Other problems that could be found, not exactly linked to accessibility are: 
- Broken Search bar; 
- CSS and JavaScript being written inside the HTML (best practice is to write it in a different document);
- In line 65, is declared a CSS file for the Search Bar that cannot be found. 
- In line 123, there's what should be a button for Google Sharing, but it is also broken. 
### The HTML element to be changed

```html
HTML declaration without stating the language: 
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
"http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">

Not using semantic tags such as <header> and <nav>
<div class="header-wrapper">
      <div class="header">
          <div class="logo"><a href="index.html">
            <img class="logo" src="_static/IPy_header.png" alt="Logo"/>
          </a></div>
       
        <div class="rel">
          <a href="https://jupyter.readthedocs.io/en/latest/install.html" target="_blank">Install</a> · 
<a href="documentation.html">Documentation</a> ·
<a href="project.html">Project</a> ·
<a href="https://jupyter.org/" target="_blank">Jupyter</a> · 
<a href="news.html">News</a> · 
<a href="citing.html">Cite</a> ·
<a href="donate.html">Donate</a> ·
<a href="books.html">Books</a>
        </div>
       </div>
    </div>

Low contrast links: 
a {
    color: #ce5c00;
}

Justified text: 
div.body {
    padding-right: 2em;
    text-align: justify;
}
```

### Your proposed HTML (or CSS), after change

```html
HTML declaration fixed: 
<!DOCTYPE html>
<html lang="en">

Fixed meta charset tag and added meta viewport:
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=Edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

Fixed semantic tags: 
<header>
<div class="logo-container">
<a href="index.html">
<img class="logo" src="img/IPy_header.png" alt="IPython's logo linking to IPython's homepage">
</a>
</div>

<nav>
<a href="https://jupyter.readthedocs.io/en/latest/install.html" target="_blank">Install</a>
<a href="documentation.html">Documentation</a>
<a href="project.html">Project</a>
<a href="https://jupyter.org/" target="_blank">Jupyter</a>
<a href="news.html">News</a>
<a href="citing.html">Cite</a> 
<a href="donate.html">Donate</a>
<a href="books.html">Books</a>
</nav>
</header>

Fixed Contrast: 
a{
    color: #2C558C;
}

Fixed Justified text: 
* {
    line-height: 1.5 !important;
    letter-spacing: 0.10em !important;
    word-spacing: 0.10em !important;
}

Added queries to adjust the content with the page: 
@media only screen and (min-width: 600px){
    header nav{
        font-size: 1.3em
    }
    main h1{
        font-size: 1.7em;
    }
    main p, ul{
        font-size: 1.5em;
    }
    aside .container{ 
        grid-template-columns: 1fr 1fr 1fr;
        grid-column-gap: 2%;
        
    }
}

@media only screen and (min-width: 1000px){
    header nav{
        font-size: 1.2em;
    }
    main h1{
        font-size: 1.5em;
    }
    main p, ul{
        font-size: 1em;
    }
    .container{
        grid-template-columns: 1fr 1fr 1fr;
        grid-column-gap: 2%;
    }
    footer .container{
        grid-template-columns: 1fr 10px 1fr;
        grid-column-gap: 0;
        justify-items: center;
    }
}
```

### More info

One of the main issues was how the screen reader started reading the sidebar content, instead of the main content first. I tried to make a recreation of the page, focused on the main structure of the page, such as tags, blocks of content and where would I put it depending on the device being used. 
Here is the link for the page: https://bumblebib.github.io/IPython_Accessibility/
And here's also the files: https://github.com/bumblebib/IPython_Accessibility

**Comment by minrk** [comment]
Great job on the mock-up website! That's very useful.

**Comment by sgibson91** [comment]
Thank you for your contribution! Since the contribution period is now over, we will close this issue.
